src/main/java/io/supertokens/bulkimport/BulkImportUserUtils.java

```diff
@@ -0,0 +1,421 @@
+            if (validateJsonFieldType(jsonLoginMethodObj, "isPrimary", ValueType.BOOLEAN)) {
+                if (jsonLoginMethodObj.get("isPrimary").getAsBoolean()) {
+                    if (hasPrimaryLoginMethod) {
+                        errors.add("No two loginMethods can have isPrimary as true.");
+                    }
+                    hasPrimaryLoginMethod = true;
+                }
+            }
+        }
+    }
+
+    private static String validateAndNormaliseRecipeId(String recipeId, List<String> errors) {
+        if (recipeId == null) {
+            return null;
+        }
+
+        // We don't perform any normalisation on the recipeId after reading it from request header.
+        // We will validate it as is.
+        if (!Arrays.asList("emailpassword", "thirdparty", "passwordless").contains(recipeId)) {
+            errors.add("Invalid recipeId for loginMethod. Pass one of emailpassword, thirdparty or, passwordless!");
+        }
+        return recipeId;
+    }
+
+    private static String validateAndNormaliseTenantId(Main main, AppIdentifier appIdentifier, String tenantId, String recipeId, List<String> errors)
+            throws StorageQueryException, TenantOrAppNotFoundException {
+        if (tenantId == null || tenantId.equals(TenantIdentifier.DEFAULT_TENANT_ID)) {
+            return tenantId;
+        }
+
+        if (Arrays.stream(FeatureFlag.getInstance(main, appIdentifier).getEnabledFeatures())
+                .noneMatch(t -> t == EE_FEATURES.MULTI_TENANCY)) {
+            errors.add("Multitenancy must be enabled before importing users to a different tenant.");
+            return tenantId;
+        }
+ 
+        // We make the tenantId lowercase while parsing from the request in WebserverAPI.java
+        String normalisedTenantId = tenantId.trim().toLowerCase();
+        TenantConfig[] allTenantConfigs = Multitenancy.getAllTenantsForApp(appIdentifier, main);
+        Set<String> validTenantIds = new HashSet<>();
+        Arrays.stream(allTenantConfigs)
+                .forEach(tenantConfig -> validTenantIds.add(tenantConfig.tenantIdentifier.getTenantId()));
+
+        if (!validTenantIds.contains(normalisedTenantId)) {
+            errors.add("Invalid tenantId: " + tenantId + " for " + recipeId + " recipe.");
+        }
+        return normalisedTenantId;
+    }
+
+    private static Boolean validateAndNormaliseIsPrimary(Boolean isPrimary) {
+        // We set the default value as false
+        return isPrimary == null ? false : isPrimary;
+    }
+
+    private static Boolean validateAndNormaliseIsVerified(Boolean isVerified) {
+        // We set the default value as false
+        return isVerified == null ? false : isVerified;
+    }
+
+    private static long validateAndNormaliseTimeJoined(Integer timeJoined, List<String> errors) {
+        // We default timeJoined to 0 if it is null
+        if (timeJoined == null) {
+            return 0;
```

Comment:
why 0? Should be the current time right?

Reply:
Fixed it to return current time.